Microsoft.NET.Build.Containers/LocalDocker.cs

```diff
@@ -29,6 +30,11 @@ public static async Task Load(Image x, string name, string tag, string baseName)
         loadProcess.StandardInput.Close();
 
         await loadProcess.WaitForExitAsync();
+
+        if (loadProcess.ExitCode != 0)
+        {
+            throw new InvalidOperationException($"Failed to load image to local Docker daemon.{Environment.NewLine}stdout: {await loadProcess.StandardError.ReadToEndAsync()}{Environment.NewLine}stderr: {await loadProcess.StandardOutput.ReadToEndAsync()}");
```

Comment:
how does this report to the user in the targets? do they need to try/catch so that ugly multiline errors don't flow through in the task reporting?

Reply:
Thanks for asking! Apparently it's just . . . ignored? Please stand by.

Reply:
Definitely not pretty. Think we should fix now or in 0.2?

```sh-session
❯ dotnet publish --os linux --arch x64 -p:PublishProfile=DefaultContainer -bl
MSBuild version 17.4.0-preview-22368-02+c8492483a for .NET
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\MSBuild.dll -bl -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\dotnet.dll -maxcpucount -property:_IsPublishing=true -property:RuntimeIdentifier=linux-x64 -property:SelfContained=false -p:PublishProfile=DefaultContainer -restore -target:Publish -verbosity:m .\containerized.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [S:\play\containerized\containerized.csproj]
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\containerized.dll
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\publish\
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: The "CreateNewImage" task failed unexpectedly. [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: System.AggregateException: One or more errors occurred. (Failed to load image to local Docker daemon. [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: stdout: unsupported os linux [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:  [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: stderr: ) [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:  ---> System.InvalidOperationException: Failed to load image to local Docker daemon. [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: stdout: unsupported os linux [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:  [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: stderr:  [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at Microsoft.NET.Build.Containers.LocalDocker.Load(Image x, String name, String tag, String baseName) [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    --- End of inner exception stack trace --- [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions) [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken) [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at System.Threading.Tasks.Task.Wait() [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at Microsoft.NET.Build.Containers.Tasks.CreateNewImage.Execute() [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [S:\play\containerized\containerized.csproj]
```

Reply:
How much of a pain is it to do now?

Reply:
Latest push does

```sh-session
❯ dotnet publish --os linux --arch x64 -p:PublishProfile=DefaultContainer -bl
MSBuild version 17.4.0-preview-22368-02+c8492483a for .NET
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\MSBuild.dll -bl -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\dotnet.dll -maxcpucount -property:_IsPublishing=true -property:RuntimeIdentifier=linux-x64 -property:SelfContained=false -p:PublishProfile=DefaultContainer -restore -target:Publish -verbosity:m .\containerized.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-su
pport-policy [S:\play\containerized\containerized.csproj]
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\containerized.dll
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\publish\
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error : Failed to load image to local Docker daemon. [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error : stdout: unsupported os linux [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error :  [S:\play\containerized\containerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error : stderr:  [S:\play\containerized\containerized.csproj]
```

I'm gonna simplify to only capture stdout since that's evidently Where It Goes™️.

Reply:
It's now

```sh-session
❯ dotnet publish --os linux --arch x64 -p:PublishProfile=DefaultContainer
MSBuild version 17.4.0-preview-22368-02+c8492483a for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-su
pport-policy [S:\play\containerized\containerized.csproj]
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\containerized.dll
  containerized -> S:\play\containerized\bin\Debug\net7.0\linux-x64\publish\
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error : Failed to load image to local Docker daemon. stdout: unsupported os linux [S:\play\containerized\contai
nerized.csproj]
C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.1.0\build\Microsoft.NET.Build.Containers.targets(66,9): error :  [S:\play\containerized\containerized.csproj]
```

Reply:
That's a huge improvement!